### DISCUSSION TITLE
Would it make sense to compile this using cosmopolitan?

### DISCUSSION THREAD

**Discussion by mofosyne** [casey/just#2355] [Ideas]
https://github.com/jart/cosmopolitan would allow this to be operated as one binary on all major system

**Comment by casey** [comment]
As far as  I know, cosmopolitan allows compiling C into cross-system binaries. `just` is written in Rust, and it would probably take a lot of tweaking to get it working with cosmopolitan.

**Comment by runeimp** [comment]
@mofosyne I believe `just` is already a single binary on all systems, no? Or was it broken down to an executable and DLL for Windows or some such?

**Comment by mofosyne** [comment]
@runeimp it's still just separate binaries between linux, mac and windows.

At the moment, rust support is still a work in progress as I can see in https://github.com/ahgamut/rust-ape-example @ahgamut managed to get it working only on x86-64-linux and aarch64-linux at this stage due to variables like `EINVAL` being only available during compile time.

So until someone figures that bit out, I think this issue ticket would remain closed. I'll ping them to let just know if it changes.

**Comment by runeimp** [comment]
Oh! A universal binary! I totally overlooked that point when looking at the Cosmopolitan GitHub the other day. My bad. 😇

**Comment by SerJaimeLannister** [comment]
what is the progress on this issue?
I would like to see just additionally supporting cosmopolitan

**Reply by casey** [reply]
Someone needs to try compiling just with cosmopolitan. I assume there will be tons of issues.
